newIDE/app/src/MainFrame/index.js

```diff
@@ -415,136 +494,124 @@ class MainFrame extends React.Component<Props, State> {
-      .then(() => this.setState({ loadingProject: false }));
-  };
-
-  closeApp = (): void => {
-    return Window.quit();
-  };
-
-  closeProject = (): Promise<void> => {
-    const { currentProject } = this.state;
-    const { eventsFunctionsExtensionsState } = this.props;
-    if (!currentProject) return Promise.resolve();
 
-    return new Promise(resolve => {
-      this.openProjectManager(false);
-      this.setState(
-        {
-          editorTabs: closeProjectTabs(this.state.editorTabs, currentProject),
-        },
-        () => {
-          eventsFunctionsExtensionsState.unloadProjectEventsFunctionsExtensions(
-            currentProject
-          );
-          currentProject.delete();
-          this.setState(
+          const serializedProject = gd.Serializer.fromJSObject(content);
+          return loadFromSerializedProject(
+            serializedProject,
+            // Note that fileMetadata is the original, unchanged one, even if we're loading
+            // an autosave. If we're for some reason loading an autosave, we still consider
+            // that we're opening the file that was originally requested by the user.
+            fileMetadata,
             {
-              currentProject: null,
-              isPreviewFirstSceneOverriden: false,
-              previewFirstSceneName: '',
+              currentProject: state.currentProject,
+              editorTabs: state.editorTabs,
+            },
+            eventsFunctionsExtensionsState
+          ).then(
+            state => {
+              serializedProject.delete();
+              return Promise.resolve(state);
             },
-            () => {
-              this.updateToolbar();
-              resolve();
+            err => {
+              serializedProject.delete();
+              throw err;
             }
           );
-        }
-      );
+        })
+        .catch(error => {
+          const errorMessage = getOpenErrorMessage
+            ? getOpenErrorMessage(error)
+            : t`Check that the path/URL is correct, that you selected a file that is a game file created with GDevelop and that is was not removed.`;
+          showErrorBox(
+            [i18n._(t`Unable to open the project.`), i18n._(errorMessage)].join(
+              '\n'
+            ),
+            error
+          );
+        });
     });
   };
 
-  getSerializedElements = () => {
-    const editorTab = getCurrentTab(this.state.editorTabs);
-    if (!editorTab || !editorTab.editorRef) {
-      console.warn('No active editor or reference to the editor');
-      return {};
-    }
-
-    return editorTab.editorRef.getSerializedElements();
-  };
-
-  toggleProjectManager = () => {
-    if (!this.refs.toolbar)
-      this.setState({
-        projectManagerOpen: !this.state.projectManagerOpen,
-      });
+  const closeApp = (): void => {
+    return Window.quit();
   };
 
-  openProjectManager = (open: boolean = true) => {
-    this.setState({
+  const closeProject = (
+    newState = {
```

Comment:
This method is never used without params. Can you entirely remove the default arguments? Could you also check all the other methods?

The idea would be to never have default arguments. Why? Because it makes type checking less efficient. 
If you forget to pass arguments:
* if arguments are mandatory, you're getting an alert. Fine!
* if arguments are not mandatory, it's maybe ok if you want the old state... but if you need the new, you don't get an alert... so you have a hard to debug issue, both for your and the reviewer. Dangerous! 
Also make harder to understand when reading the code "why the heck do they sometimes pass the project, and sometimes not?? The method does not always need the project??"

Conclusion:
let's not have default arguments. Instead, let's be explicit and always pass what is needed.

Advantage: we'll progressively remove the notion of "newState". It's not helping to understand a method if I tell someone "closeProject is taking an argument called newState" ("ok sure, and so what it this thing huh?"). But if I tell "closeProject is taking currentProject and editorTabs", it's slightly interesting ("ok, so the function needs to deal with the project and the tabs - make sense I guess").